
<!DOCTYPE>
<html lang="en" data-scribe-reduced-action-queue="true">
  <head>
    <meta content='PROTREC - An algorithm for predicting and validating missing proteins in proteomics - Centre for Biomedical Informatics | NTU Singapore' name='title' />
    <meta content='PROTREC - An algorithm for predicting and validating missing proteins in proteomics - Centre for Biomedical Informatics | NTU Singapore' name='og:title' />
    <title>PROTREC - An algorithm for predicting and validating missing proteins in proteomics - Centre for Biomedical Informatics | NTU Singapore</title>
    <!--                                                                                                                                                    
                               bbbbbbbb                                                                                                            
LLLLLLLLLLL               iiii b::::::b                                                 XXXXXXX       XXXXXXX                                      
L:::::::::L              i::::ib::::::b                                                 X:::::X       X:::::X                                      
L:::::::::L               iiii b::::::b                                                 X:::::X       X:::::X                                      
LL:::::::LL                     b:::::b                                                 X::::::X     X::::::X                                      
  L:::::L               iiiiiii b:::::bbbbbbbbb        eeeeeeeeeeee    rrrrr   rrrrrrrrrXXX:::::X   X:::::XXXuuuuuu    uuuuuu      eeeeeeeeeeee    
  L:::::L               i:::::i b::::::::::::::bb    ee::::::::::::ee  r::::rrr:::::::::r  X:::::X X:::::X   u::::u    u::::u    ee::::::::::::ee  
  L:::::L                i::::i b::::::::::::::::b  e::::::eeeee:::::eer:::::::::::::::::r  X:::::X:::::X    u::::u    u::::u   e::::::eeeee:::::ee
  L:::::L                i::::i b:::::bbbbb:::::::be::::::e     e:::::err::::::rrrrr::::::r  X:::::::::X     u::::u    u::::u  e::::::e     e:::::e
  L:::::L                i::::i b:::::b    b::::::be:::::::eeeee::::::e r:::::r     r:::::r  X:::::::::X     u::::u    u::::u  e:::::::eeeee::::::e
  L:::::L                i::::i b:::::b     b:::::be:::::::::::::::::e  r:::::r     rrrrrrr X:::::X:::::X    u::::u    u::::u  e:::::::::::::::::e 
  L:::::L                i::::i b:::::b     b:::::be::::::eeeeeeeeeee   r:::::r            X:::::X X:::::X   u::::u    u::::u  e::::::eeeeeeeeeee  
  L:::::L         LLLLLL i::::i b:::::b     b:::::be:::::::e            r:::::r         XXX:::::X   X:::::XXXu:::::uuuu:::::u  e:::::::e           
LL:::::::LLLLLLLLL:::::Li::::::ib:::::bbbbbb::::::be::::::::e           r:::::r         X::::::X     X::::::Xu:::::::::::::::uue::::::::e          
L::::::::::::::::::::::Li::::::ib::::::::::::::::b  e::::::::eeeeeeee   r:::::r         X:::::X       X:::::X u:::::::::::::::u e::::::::eeeeeeee  
L::::::::::::::::::::::Li::::::ib:::::::::::::::b    ee:::::::::::::e   r:::::r         X:::::X       X:::::X  uu::::::::uu:::u  ee:::::::::::::e  
LLLLLLLLLLLLLLLLLLLLLLLLiiiiiiiibbbbbbbbbbbbbbbb       eeeeeeeeeeeeee   rrrrrrr         XXXXXXX       XXXXXXX    uuuuuuuu  uuuu    eeeeeeeeeeeeee                                                                                                                                                                                                                                                                                                   
!-->
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
  <title>PROTREC - An algorithm for predicting and validating missing proteins in proteomics - Centre for Biomedical Informatics | NTU Singapore</title>
  <meta name="twitter:card" content="summary">
  <meta name="twitter:title" content="PROTREC - An algorithm for predicting and validating missing proteins in proteomics - Centre for Biomedical Informatics | NTU Singapore">
  <meta name="twitter:keywords"  content="PROTREC - An algorithm for predicting and validating missing proteins in proteomics - Centre for Biomedical Informatics | NTU Singapore" property='og:description' />
  <meta property="og:type" content="article">
  <meta property="og:title" content="PROTREC - An algorithm for predicting and validating missing proteins in proteomics - Centre for Biomedical Informatics | NTU Singapore">
  <meta name="og:keywords"  content="PROTREC - An algorithm for predicting and validating missing proteins in proteomics - Centre for Biomedical Informatics | NTU Singapore|PROTREC PROTEC is an algorithm for predicting and validating missing proteins in proteomics data based on Kong W, Won..." property='og:description' />
  <meta name="Centre for Biomedical Informatics" content="#eb5055"> 
  <link rel="icon" type="image/png" href="https://cbi-ntu.github.io/CBI-Project-Website.github.io/style/favicons/favicon.ico" />
  <link href="https://cbi-ntu.github.io/CBI-Project-Website.github.io/style/favicons/favicon.ico" rel="shortcut icon" type="image/png">
  <link rel="canonical" href="https://cbi-ntu.github.io/CBI-Project-Website.github.io/">
  <link rel="alternate" type="application/rss+xml" title="CBI" href="https://cbi-ntu.github.io/CBI-Project-Website.github.io/feed.xml">
  <meta charset="UTF-8">  
  <link rel="manifest" href="/manifest.json">
  <link rel="shortcut icon" href="https://cbi-ntu.github.io/CBI-Project-Website.github.io/style/favicons/favicon.ico" type="image/x-icon">
  <meta name="keywords"  content="PROTREC PROTEC is an algorithm for predicting and validating missing proteins in proteomics data based on Kong W, Won..." property='og:description' />
  <meta name="description" content="nothing is more practicla than a good theory" />
  <link href="https://cdn.bootcss.com/highlight.js/9.10.0/styles/xcode.min.css" rel="stylesheet">
  <link href="https://cbi-ntu.github.io/CBI-Project-Website.github.io/style/style-cbi.css" rel="stylesheet">

  <meta content='https://cbi-ntu.github.io/CBI-Project-Website.github.io/2018/07/30/wj.html' property='og:url' />
  <meta content="https://cbi-ntu.github.io/CBI-Project-Website.github.io/2018/07/30/wj.html|PROTREC PROTEC is an algorithm for predicting and validating missing proteins in proteomics data based on Kong W, Won..." property='og:description' />
  <meta content="article" property="og:type" />



<script src="https://cbi-ntu.github.io/CBI-Project-Website.github.io/script/jquery/jquery.js"></script>
<script async src="https://cbi-ntu.github.io/CBI-Project-Website.github.io/script/mathjax/tex-chtml.js"></script>
  </head>
<body class="" gtools_scp_screen_capture_injected="true">
<header id="header" class="header bg-white">
    <div class="navbar-container">
      <a href="" title="Find out more about Centre for Biomedical Informatics | NTU Singapore" class="navbar-logo"> <img src="https://cbi-ntu.github.io/CBI-Project-Website.github.io/media/logo.png" alt="Centre for Biomedical Informatics | NTU Singapore"> </a>
        <div class="navbar-menu">
            
                <a href="https://cbi-ntu.github.io/CBI-Project-Website.github.io/about">About CBI</a> 
                 
                <a href="https://cbi-ntu.github.io/CBI-Project-Website.github.io/">Our Projects</a> 
                 
                <a href="https://cbi-ntu.github.io/CBI-Project-Website.github.io/services">Our Services</a> 
                 
                <a href="https://cbi-ntu.github.io/CBI-Project-Website.github.io/contact">Contact Us</a> 
                 
          </div>
        <div class="navbar-search" onclick="">
            <span class="icon-search"></span>
            <form id="cb-search-btn" role="search">
         <span class="search-box">
         <input type="text" class="input" id="cb-search-content" required="true" placeholder="Contact Us..." maxlength="30" autocomplete="off">
        </span>
            </form>
        </div>
        <div class="navbar-mobile-menu" onclick="">
            <span class="icon-menu cross"><span class="middle"></span></span>
            <ul>
                
                <li><a href="https://cbi-ntu.github.io/CBI-Project-Website.github.io/about">About CBI</a></li> 
                 
                <li><a href="https://cbi-ntu.github.io/CBI-Project-Website.github.io/">Our Projects</a></li> 
                 
                <li><a href="https://cbi-ntu.github.io/CBI-Project-Website.github.io/services">Our Services</a></li> 
                 
                <li><a href="https://cbi-ntu.github.io/CBI-Project-Website.github.io/contact">Contact Us</a></li> 
                 
            </ul>
        </div>
    </div>
</header>
<div class="post-header-thumb bg-white}">
	<div class="post-header-thumb-op"></div>
	<div class="post-header-thumb">
		<div class="post-header-thumb-container">
    <h1 class="post-title" itemprop="name headline">
	PROTREC - An algorithm for predicting and validating missing proteins in proteomics</h1>
		 <div class="post-data">  
	 <time datetime="2018-07-30 10:06:42" itemprop="datePosted">Date Posted: 2018-07-30 10:06:42</time>
     <a href="https://cbi-ntu.github.io/CBI-Project-Website.github.io/tags#Cancer" title="Find out more about Cancer" data-hover="Category: Cancer">Category: Cancer</a>
    <a href="#read"> Reads: comments</a> 
			</div>
            <div class="post-tags">
                
      
      
       
          <a href="https://cbi-ntu.github.io/CBI-Project-Website.github.io/tags#ML" title="Find out more aboutML" data-hover="ML">
          ML <span>(2)</span>
          
</a>
       
          <a href="https://cbi-ntu.github.io/CBI-Project-Website.github.io/tags#Cancer" title="Find out more aboutCancer" data-hover="Cancer">
          Cancer <span>(2)</span>
          
</a>
       
          <a href="https://cbi-ntu.github.io/CBI-Project-Website.github.io/tags#Explainability" title="Find out more aboutExplainability" data-hover="Explainability">
          Explainability <span>(2)</span>
          
</a>
      
      
      
     
	</div>
</div>
</div></div>
<article class="main-content post-page" itemscope itemtype="http://schema.org/Article">
	<div class="post-header">
		<h1 class="post-title" itemprop="name headline">
			PROTREC - An algorithm for predicting and validating missing proteins in proteomics</h1>
		<div class="post-data">
				<time datetime="2018-07-30 10:06:42" itemprop="datePosted">2018-07-30 10:06:42</time>
		</div>
	</div>
	<div id="post-content" class="post-content" itemprop="articleBody">
         <p class="post-tags">
    
      
      
       
          <a href="https://cbi-ntu.github.io/CBI-Project-Website.github.io/tags#ML" title="Find out more aboutML" data-hover="ML">
          ML <span>(2)</span>
          </a>
       
          <a href="https://cbi-ntu.github.io/CBI-Project-Website.github.io/tags#Cancer" title="Find out more aboutCancer" data-hover="Cancer">
          Cancer <span>(2)</span>
          </a>
       
          <a href="https://cbi-ntu.github.io/CBI-Project-Website.github.io/tags#Explainability" title="Find out more aboutExplainability" data-hover="Explainability">
          Explainability <span>(2)</span>
          </a>
      
      
      
     
</p>
<h1 id="protrec">PROTREC</h1>
<p><img src="https://cbi-ntu.github.io/CBI-Project-Website.github.io/media/wj_abstract.png" /> PROTEC is an algorithm for predicting and validating missing proteins in proteomics data based on <a href="https://www.sciencedirect.com/science/article/pii/S1874391921002918">Kong W, Wong B J H, Gao H, et al. PROTREC: A probability-based approach for recovering missing proteins based on biological networks. Journal of Proteomics, 2022, 250: 104392.</a> Here, we also include other three common used network based methods (same as in the literature) for a better comparison.</p>

<div class="h30"></div>
<div class="col_w310 float_l">
    <h3>Functions our platform provide</h3>
    <ul>
        <li>Probability (or p-value) score matrix</li>
        <li>Comparison heatmap</li>
        <li>Recovery rate</li>
        <li>Score distribution</li>
    </ul>
</div>
<div class="col_w310 float_r">
    <h3>Methods included in this platform</h3>
    <ul>
        <li> PROTREC </li>
        <li> FCS (Functional Class Scoring) </li>
        <li> HE (Hypergeometic Enrichment) </li>
        <li> GSEA (Gene Set Enrichment Analysis) </li>
    </ul>
</div>
<div class="cleaner"></div>

<h2 id="methods-description">Methods Description</h2>
<h3 id="protrec-1">PROTREC</h3>
<p> PROTREC is a novel probability-based scoring scheme that estimates the probability of a protein being present in a screen. </p>

<p>Specifically, to calculate PROTREC probability for a protein 
$(x)$, we first find all the protein complexes $(z_i)$ containing protein $(x)$. Not all complex we should consider since some complex contains so less proteins. <b><u>The default complex size threshold is 5, you may change it by changing "Min size of protein complex" in the parameter setting under "Run your own datasets" section.</u></b></p>

<p>Then, we calculate the probability of a complex $(z_i\in z)$ being present: 
<script type="math/tex">\[p(z_i)=\frac{\sum\limits_{x_i\in L}(1-FDR)}{|z_i|}\]\(x_i\)</script></p>

<p>denotes a protein inside the complex $(z_i)$. $(L)$ denotes the set of proteins reported by the proteomic screen. &lt;/p&gt;</p>

<p>If \(x_i\) is reported by the proteomics screen \(L\), then its prior probability is \((1-FDR)\), where \(FDR\) is the false discovery rate of \(L\). <u><b>The default \(FDR\) is 0.01, you may change it by changing "FDR of proteomic screen" in the parameter setting under "Run your own datasets" section.</b></u>
<p>PROTREC assume protein \(x\) being present in a sample is depe$ndent on the joint probability of it being present if its complex is formed, and the probability it is present if its constituent complex is not formed. Since there might be multiple protein complexes, PROTREC computes the probability of a protein \(x\) being present in a sample being screened using each of the complexes that the protein \(x\) is a member of and returns the maximum:\[p(x)=\max_{z_i\in z}\{p(x|z_i)p(z_i)+p(x|\overline{z_i})p(\overline{z_i})\}\]This way, we can calculate all protein's PROTREC score. We can sort the proteins by their score and predict unreported proteins above a given PROTREC score threshold as predicted missing proteins. <b><u>By default, we use 0.95 as the cutoff. You may change your own cutoff by changing `PROTREC score threshold' in the parameter setting under `Run your own datasets' section.</u></b></p>

### FCS
<p>FCS tests whether a network is significantly enriched given the observed proteins. Given a set of observed proteins in a proteomics screen \(S\), and a list of component proteins \(M\) from protein complex \(C\), an observed overlap \(O\), which is expressed as: \[O=\frac{|S\cap M|}{|M|}\]To determine if the overlap \(O\) is significant, a set \(N\) of randomized complexes of size are generated using a reference pool of unique proteins drawn from the complexes \(C\). <b><u>Default we choose \(N\) as 1000, but you may change the number by changing `Number of iterations' in the parameter setting under `Run your own datasets' section.</u></b> Among the randomized complexes, a vector of null overlaps, \(N_j\) is generated. For the \(j^{th}\) randomized complex, which comprises the set of proteins \(K_j\), \(N_j\), is defined as follows:\[N_j=\frac{|S\cap K_j|}{|K_j|}\]The empirical p-value is the proportion of null overlaps in \(N_j\) greater than or equal to the observed overlap \(O\). For the \(i^{th}\) complex \(C_i\) in the complex vector, its p-value, \(pval_i\) is:\[pval_i=\frac{\sum\limits_{j=1}^{N}[N_j\ge 0]}{N}\]If the FCS p-value falls below a significance p-value threshold, then all member proteins of the complex, including the unobserved ones, are predicted as present. </p>

### HE
<p>In HE, the set of observed proteins are compared against a vector of protein complexes. Given a total number of proteins \(N\), with \(M\) of these belonging to a complex and \(n\) of these proteins in the differential set, the probability \(P\) that \(b\) or more proteins from the differential set are associated by chance with the complex is given by:\[P(X\ge b)=\sum\limits_{i=b}^{min(n,M)}\frac{C_n^iC_{N-n}^{M-i}}{C_N^M}\]\(P(X\ge b)\) is the HE p-value. A complex is declared significant if the HE p-value falls below the threshold.</p>


### GSEA
<p>Gene Set Enrichment Analysis (GSEA) uses a Kolmogorov-Smirnov (KS) statistic. Here, the two-sample KS test is used to evaluate if the distribution of ranks based on the t-statistic of proteins in a complex differs from that of proteins outside the complex. Denoting proteins in the complex as the set \(C\) and proteins outside the complex as the set \(C'\), the KS-statistic \(KS_{C,C'}\)is expressed as:\[KS_{C,C'}=\max\limits_x|F_{1,C}(x)-F_{2,C'}(x)|\]where \(F_{1,C}(x)\) and \(F_{2,C'}(x)\) are respectively the fraction of proteins in \(C\) and \(C'\) whose rank is higher than the rank \(x\). The null hypothesis is rejected at a significance threshold if\[KS_{C,C'}\ge c(\alpha)*\sqrt{\frac{|C|+|C'|}{|C|*|C'|}}\]Where \(c(\alpha)\) is the critical value at a given alpha level. </p>

<p>Notably, <b><u>For FCS, HE and GSEA, the default p-value threshold is 0.05. You may change your own cutoff by changing `p-val cutoff for FCS, HE and GSEA' in the parameter setting under `Run your own datasets' section.</u></b></p>

## Example
### Example dataset: RC_N data
<p>One proteomics expression datasets are provided. The renal cancer dataset (RC) comprises 12 normal (RC_N) and 12 cancer (RC_C) samples. We provide RC_N sample as an example. Press the button you may check how does the <a href="/static/csv/RCN.csv" download="RCN.csv">sample</a> looks like.</p>
<br />

### Protein complex
<p>Protein complex is obtained from <a href="http://mips.helmholtz-muenchen.de/genre/proj/corum/">CORUM</a>. We use CORUM complex 2018 release human dataset as our default complex database. You may check how the complex looks like by <a href="/static/csv/complex.csv" download="complex.csv">clicking here</a></p>

<br />

### Example results
<ul>
    <li><a href="/static/csv/example_protrecres.csv" download="example_protrecres.csv">PROTREC result</a></li>
    <li><a href="/static/csv/example_fcsres.csv" download="example_fcsres.csv">FCS result</a></li>
    <li><a href="/static/csv/example_heres.csv" download="example_heres.csv">HE result</a></li>
    <li><a href="/static/csv/example_gseares.csv" download="example_gseares.csv">GSEA result</a></li>
</ul>
                
            

## Submit youur dataset
<p> You can submit your own dataset and protein complex to get the protein inference result. If it is your first time using this tool, you may refer to <a href="#">Example</a> section to see how it works. If no file is submitted, the program will use default file. We encourage users to use the default protein complex.</p>



#### Step 1. Upload your Dataset and Protein Complex
<p> Dataset (Default: example dataset)</p>
<div>
    <input id="datasetSelector" type="file" accept=".csv" />
    <div class="h10"></div>
    <button onclick="chooseFile('datasetSelector')">Upload</button>
    <div class="h20"></div>
</div>
<p id="datasetHint" class="hint"></p>
<p>Protein Complex (Default: CORUM database 2018 release)</p>
<div>
    <input id="complexSelector" type="file" accept=".csv" />
    <div class="h10"></div>
    <button onclick="chooseFile('complexSelector')">Upload</button>
</div>
<p id="complexHint" class="hint"></p>


#### Step 2. Select the results
<b>Select the method</b>
<div class="h10"></div>
<div>
    <input type="radio" name="resultKind" /> <span class="optionSpan">PROTREC</span>
    &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
    <input type="radio" name="resultKind" /> <span class="optionSpan">ALL</span>
</div>
<div class="h20"></div>
<b>Select the results</b>
<div class="h10"></div>
<div>
    <input type="checkbox" name="resultType" /> <span class="optionSpan">Raw predicted score matrix</span>
    <div class="h10"></div>
    <input type="checkbox" name="resultType" /> <span class="optionSpan">0/1 predicted score matrix</span>
</div>
<div class="h10"></div>
<div>
    <input type="checkbox" name="ref" /> <span class="optionSpan">Recovery rate</span>
    <div class="h10"></div>
    <input type="checkbox" name="ref" /> <span class="optionSpan">Score distribution</span>
</div>
<div class="h20"></div>
<b>Note: recovery rate and score distribution can only be calculated after you submit your reference list.</b><br />
<div class="h10"></div>
Upload your reference list
<div class="h10"></div>
<div>
    <input id="refSelector" type="file" accept=".csv" />
    <div class="h10"></div>
    <button onclick="chooseFile('refSelector')">Upload</button>
    <div class="h20"></div>
</div>
<p id="refHint" class="hint"></p>

#### Step 3. Enter your parameters
<div>
    <label>Number of iterations</label><br />
    <input id="iterInput" />
</div>
<div>
    <label>Min size of protein complex</label><br />
    <input id="minsizeInput" />
</div>
<div>
    <label>FDR of proteomic screen</label><br />
    <input id="mafdrInput" />
</div>
<div>
    <label>PROTREC score threshold</label><br />
    <input id="protrecscoresetInput" />
</div>
<div>
    <label>FDR threshold for FCS, HE and GSEA</label><br />
    <input id="fdrsetInput" />
</div>
&lt;/div&gt;

#### Step 4. Enter email and key
<div>
    <label>Email Address</label> <br />
    <input id="email" />
</div>
<div>
    <label>Confirm Email Address</label> <br />
    <input id="emailConfirm" />
</div>
<div class="h10"></div>
<p>Click <a href="javascript:void(0)" onclick="getVerificationKey()">here</a> to get your verification key after putting in your email address.</p>
<div>
    <label>Verification Key</label> <br />
    <input id="key" />
</div>
<p>Click <a href="javascript:void(0)" onclick="sendEmail()">here</a> to send your data to us. We will execute your data and send results back to your email.</p>
<p id="emailHint" class="hint"></p>
&lt;/div&gt;
<div class="cleaner"></div>
&lt;/html&gt;
</p>

		<!--<p class="post-info">
			本文由 <a href="/">liberxue</a> 创作，采用 <a href="https://creativecommons.org/licenses/by/4.0/" target="_blank" rel="external nofollow">知识共享署名4.0</a> 国际许可协议进行许可<br>本站文章除注明转载/出处外，均为本站原创或翻译，转载前请务必署名<br>最后编辑时间为:2018-07-30 10:06:42</p>-->
	</div>
</article>

<!-- <div class="related-post-lists">
	<div class="post-lists">
		<div class="post-lists-body">
					<div class="post-list-item">
				<div class="post-list-item-container">
					<div class="item-label">
						<div class="item-title"><a href="induction.html">归纳的随想</a></div>
						<div class="item-meta clearfix">
														<div class="item-meta-ico bg-ico-image" style="background: url('../usr/themes/pinghsu/images/bg-ico.png') no-repeat;background-size: 40px auto;"></div>
	                        							<div class="item-meta-cat"><a href="../category/thoughts/index.html">Thoughts</a></div>
						</div>
					</div>
				</div>
			</div>
				</div>
	</div>
</div> -->
<footer class="footer bg-white">
	<div class="footer-social">
		<div class="footer-container clearfix">
			<div class="social-list">
    <!--<a class="social github" target="blank" href="https://github.com/" title="Find out more about " data-hover="GitHub">GitHub</a>
    <a class="social twitter" target="blank" href="http://twitter.com/" title="Find out more about " data-hover="Twitter">Twitter</a>
    <a class="social rss" target="blank" href="/feed.xml"title="Find out more about " data-hover="RSS">RSS</a>-->
	</div>
		</div>
	</div>
	<div class="footer-meta">
        <div class="footer-container">
            <div class="meta-item meta-copyright">
                <div class="meta-copyright-info">
                    <a href="/" class="info-logo">
                         <img src="https://cbi-ntu.github.io/CBI-Project-Website.github.io/media/logo.png"   title="Find out more about LiberXue_blog" data-hover="LiberXue_blog" alt="LiberXue_blog" /> </a>
                    <div class="info-text">
                        <!--<p>Theme is <a href="http://www.liberxue.com/2017/08/03/Jekyllthemes.html" title="Find out more about Jekyll liberxue主题"  data-hover="Jekyll liberxue"target="_blank">Jekyll liberxue</a> by <a href="http://www.liberxue.com/about" target="_blank">liberxue</a></p>-->
                        <p>Powered by <a href="https://cbi-ntu.github.io/CBI-Project-Website.github.io/tags/#Jekyll" title="Jekyll" data-hover="Jekyll" target="_blank" rel="nofollow">Jekyll</a></p>
                        <p>&copy; 2017 <a href="/feed.xml"  title="CBI blog RSS" data-hover="CBI blog RSS">CBI blog RSS</a></p>
                    </div>
                </div>
            </div>
            <div class="meta-item meta-posts">
                <h3 class="meta-title">Pinned articles</h3>
     <!--置顶-->
    
      
    
      
    
      
    
        <!--置顶-->
            </div>
            <div class="meta-item meta-comments">
                <h3 class="meta-title">Latest project</h3>
          
      
            
      
            
      
    
</div>
		</div>
    </div>
</footer>
<script src="//cdn.bootcss.com/jquery/2.2.2/jquery.min.js"></script>
<script src="//cdn.bootcss.com/bootstrap-3-typeahead/4.0.2/bootstrap3-typeahead.js"></script>
<script src="https://cbi-ntu.github.io/CBI-Project-Website.github.io/search/js/cb-search.js"></script>
<!--<script>
$(function () {
    setTimeout(function () {
      $('code').removeClass('hljs');
      $('code').removeClass('language');
    }, 90);
});
(function(){
    var bp = document.createElement('script');
    var curProtocol = window.location.protocol.split(':')[0];
    if (curProtocol === 'https') {
        bp.src = 'https://zz.bdstatic.com/linksubmit/push.js';        
    }
    else {
        bp.src = 'http://push.zhanzhang.baidu.com/push.js';
    }
    var s = document.getElementsByTagName("script")[0];
    s.parentNode.insertBefore(bp, s);
})();
</script>-->

<!--liberxueconnnet-menu-->
<div id="directory-content" class="directory-content">
    <div id="directory"></div>
</div>
<!--liberxueconnnet-menu-->
<script>
var postDirectoryBuild = function() {
    var postChildren = function children(childNodes, reg) {
        var result = [],
            isReg = typeof reg === 'object',
            isStr = typeof reg === 'string',
            node, i, len;
        for (i = 0, len = childNodes.length; i < len; i++) {
            node = childNodes[i];
            if ((node.nodeType === 1 || node.nodeType === 9) &&
                (!reg ||
                isReg && reg.test(node.tagName.toLowerCase()) ||
                isStr && node.tagName.toLowerCase() === reg)) {
                result.push(node);
            }
        }
        return result;
    },
    createPostDirectory = function(article, directory, isDirNum) {
        var contentArr = [],
            titleId = [],
            levelArr, root, level,
            currentList, list, li, link, i, len;
        levelArr = (function(article, contentArr, titleId) {
            var titleElem = postChildren(article.childNodes, /^h\d$/),
                levelArr = [],
                lastNum = 1,
                lastRevNum = 1,
                count = 0,
                guid = 1,
                id = 'directory' + (Math.random() + '').replace(/\D/, ''),
                lastRevNum, num, elem;
            while (titleElem.length) {
                elem = titleElem.shift();
                contentArr.push(elem.innerHTML);
                num = +elem.tagName.match(/\d/)[0];
                if (num > lastNum) {
                    levelArr.push(1);
                    lastRevNum += 1;
                } else if (num === lastRevNum ||
                    num > lastRevNum && num <= lastNum) {
                    levelArr.push(0);
                    lastRevNum = lastRevNum;
                } else if (num < lastRevNum) {
                    levelArr.push(num - lastRevNum);
                    lastRevNum = num;
                }
                count += levelArr[levelArr.length - 1];
                lastNum = num;
                elem.id = elem.id || (id + guid++);
                titleId.push(elem.id);
            }
            if (count !== 0 && levelArr[0] === 1) levelArr[0] = 0;

            return levelArr;
        })(article, contentArr, titleId);
        currentList = root = document.createElement('ul');
        dirNum = [0];
        for (i = 0, len = levelArr.length; i < len; i++) {
            level = levelArr[i];
            if (level === 1) {
                list = document.createElement('ul');
                if (!currentList.lastElementChild) {
                    currentList.appendChild(document.createElement('li'));
                }
                currentList.lastElementChild.appendChild(list);
                currentList = list;
                dirNum.push(0);
            } else if (level < 0) {
                level *= 2;
                while (level++) {
                    if (level % 2) dirNum.pop();
                    currentList = currentList.parentNode;
                }
            }
            dirNum[dirNum.length - 1]++;
            li = document.createElement('li');
            link = document.createElement('a');
            link.href = '#' + titleId[i];
            link.title = 'Find out more about' + titleId[i];
            link.title = 'Find out more about' + titleId[i];
            link.innerHTML = !isDirNum ? contentArr[i] :
                dirNum.join('.') + ' ' + contentArr[i] ;
            li.appendChild(link);
            currentList.appendChild(li);
        }
        directory.appendChild(root);
    };
    createPostDirectory(document.getElementById('post-content'),document.getElementById('directory'), true);
};
postDirectoryBuild();
</script>
  <script src="cdn.bootcss.com/highlight.js/9.12.0/highlight.min.js"></script>
<script >lang=hljs.initHighlightingOnLoad();</script>   
 </body>
</html>