
<!DOCTYPE>
<html lang="en" data-scribe-reduced-action-queue="true">
  <head>
    <meta content='PROTREC: a probability-based approach for recovering missing proteins based on biological networks - Centre for Biomedical Informatics | NTU Singapore' name='title' />
    <meta content='PROTREC: a probability-based approach for recovering missing proteins based on biological networks - Centre for Biomedical Informatics | NTU Singapore' name='og:title' />
    <title>PROTREC: a probability-based approach for recovering missing proteins based on biological networks - Centre for Biomedical Informatics | NTU Singapore</title>
    <!--                                                                                                                                                    
                               bbbbbbbb                                                                                                            
LLLLLLLLLLL               iiii b::::::b                                                 XXXXXXX       XXXXXXX                                      
L:::::::::L              i::::ib::::::b                                                 X:::::X       X:::::X                                      
L:::::::::L               iiii b::::::b                                                 X:::::X       X:::::X                                      
LL:::::::LL                     b:::::b                                                 X::::::X     X::::::X                                      
  L:::::L               iiiiiii b:::::bbbbbbbbb        eeeeeeeeeeee    rrrrr   rrrrrrrrrXXX:::::X   X:::::XXXuuuuuu    uuuuuu      eeeeeeeeeeee    
  L:::::L               i:::::i b::::::::::::::bb    ee::::::::::::ee  r::::rrr:::::::::r  X:::::X X:::::X   u::::u    u::::u    ee::::::::::::ee  
  L:::::L                i::::i b::::::::::::::::b  e::::::eeeee:::::eer:::::::::::::::::r  X:::::X:::::X    u::::u    u::::u   e::::::eeeee:::::ee
  L:::::L                i::::i b:::::bbbbb:::::::be::::::e     e:::::err::::::rrrrr::::::r  X:::::::::X     u::::u    u::::u  e::::::e     e:::::e
  L:::::L                i::::i b:::::b    b::::::be:::::::eeeee::::::e r:::::r     r:::::r  X:::::::::X     u::::u    u::::u  e:::::::eeeee::::::e
  L:::::L                i::::i b:::::b     b:::::be:::::::::::::::::e  r:::::r     rrrrrrr X:::::X:::::X    u::::u    u::::u  e:::::::::::::::::e 
  L:::::L                i::::i b:::::b     b:::::be::::::eeeeeeeeeee   r:::::r            X:::::X X:::::X   u::::u    u::::u  e::::::eeeeeeeeeee  
  L:::::L         LLLLLL i::::i b:::::b     b:::::be:::::::e            r:::::r         XXX:::::X   X:::::XXXu:::::uuuu:::::u  e:::::::e           
LL:::::::LLLLLLLLL:::::Li::::::ib:::::bbbbbb::::::be::::::::e           r:::::r         X::::::X     X::::::Xu:::::::::::::::uue::::::::e          
L::::::::::::::::::::::Li::::::ib::::::::::::::::b  e::::::::eeeeeeee   r:::::r         X:::::X       X:::::X u:::::::::::::::u e::::::::eeeeeeee  
L::::::::::::::::::::::Li::::::ib:::::::::::::::b    ee:::::::::::::e   r:::::r         X:::::X       X:::::X  uu::::::::uu:::u  ee:::::::::::::e  
LLLLLLLLLLLLLLLLLLLLLLLLiiiiiiiibbbbbbbbbbbbbbbb       eeeeeeeeeeeeee   rrrrrrr         XXXXXXX       XXXXXXX    uuuuuuuu  uuuu    eeeeeeeeeeeeee                                                                                                                                                                                                                                                                                                   
!-->
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
  <title>PROTREC: a probability-based approach for recovering missing proteins based on biological networks - Centre for Biomedical Informatics | NTU Singapore</title>
  <meta name="twitter:card" content="summary">
  <meta name="twitter:title" content="PROTREC: a probability-based approach for recovering missing proteins based on biological networks - Centre for Biomedical Informatics | NTU Singapore">
  <meta name="twitter:keywords"  content="PROTREC: a probability-based approach for recovering missing proteins based on biological networks - Centre for Biomedical Informatics | NTU Singapore" property='og:description' />
  <meta property="og:type" content="article">
  <meta property="og:title" content="PROTREC: a probability-based approach for recovering missing proteins based on biological networks - Centre for Biomedical Informatics | NTU Singapore">
  <meta name="og:keywords"  content="PROTREC: a probability-based approach for recovering missing proteins based on biological networks - Centre for Biomedical Informatics | NTU Singapore|AbstractA novel network-based approach for predicting missing proteins (MPs) is proposed here. This approach, PROTREC..." property='og:description' />
  <meta name="Centre for Biomedical Informatics" content="#eb5055"> 
  <link rel="icon" type="image/png" href="https://cbi-ntu.github.io/CBI-Project-Website.github.io/style/favicons/favicon.ico" />
  <link href="https://cbi-ntu.github.io/CBI-Project-Website.github.io/style/favicons/favicon.ico" rel="shortcut icon" type="image/png">
  <link rel="canonical" href="https://cbi-ntu.github.io/CBI-Project-Website.github.io/">
  <link rel="alternate" type="application/rss+xml" title="CBI" href="https://cbi-ntu.github.io/CBI-Project-Website.github.io/feed.xml">
  <meta charset="UTF-8">  
  <link rel="manifest" href="/manifest.json">
  <link rel="shortcut icon" href="https://cbi-ntu.github.io/CBI-Project-Website.github.io/style/favicons/favicon.ico" type="image/x-icon">
  <meta name="keywords"  content="AbstractA novel network-based approach for predicting missing proteins (MPs) is proposed here. This approach, PROTREC..." property='og:description' />
  <meta name="description" content="nothing is more practicla than a good theory" />
  <link href="https://cdn.bootcss.com/highlight.js/9.10.0/styles/xcode.min.css" rel="stylesheet">
  <link href="https://cbi-ntu.github.io/CBI-Project-Website.github.io/style/style-cbi.css" rel="stylesheet">

  <meta content='https://cbi-ntu.github.io/CBI-Project-Website.github.io/2022/01/01/bookindex.html' property='og:url' />
  <meta content="https://cbi-ntu.github.io/CBI-Project-Website.github.io/2022/01/01/bookindex.html|AbstractA novel network-based approach for predicting missing proteins (MPs) is proposed here. This approach, PROTREC..." property='og:description' />
  <meta content="article" property="og:type" />

  <!--test insert start-->
    <script src="static/js/jquery/jquery.js"></script>
        <script async src="static/js/mathjax/tex-chtml.js"></script>
        <script>
    "use strict";
    var datasetData = null;
    var complexData = null;
    var refData = null;
    var chooseFile = function (id) {
        let filePath = $(`#${id}`).get(0).files[0];
        if (!filePath) {
            if (id === "datasetSelector") {
                $("#datasetHint").text("Dataset file not selected").css("color", "red").show();
            } else if (id === "complexSelector") {
                $("#complexHint").text("Complex file not selected").css("color", "red").show();
            } else if (id === "refSelector") {
                $("#refHint").text("Reference List file not selected").css("color", "red").show();
            }
            return;
        }
        let reader = new FileReader ();
        if (id === "datasetSelector") {
            reader.onloaderror = function () {
                $("#datasetHint").text(`Error on loading ${filePath.name}!`).css("color", "red").show();
            };
            reader.onload = function () {
                $("#datasetHint").text(`Finish loading ${filePath.name}`).css("color", "blue").show();
                datasetData = reader.result;
            }
        } else if (id === "complexSelector") {
            reader.onloaderror = function () {
                $("#complexHint").text(`Error on loading ${filePath.name}!`).css("color", "red").show();
            }
            reader.onload = function () {
                $("#complexHint").text(`Finish loading ${filePath.name}`).css("color", "blue").show();
                complexData = reader.result;
            }
        } else if (id === "refSelector") {
            reader.onloaderror = function () {
                $("#refHint").text(`Error on loading ${filePath.name}!`).css("color", "red").show();
            }
            reader.onload = function () {
                $("#refHint").text(`Finish loading ${filePath.name}`).css("color", "blue").show();
                refData = reader.result;
                $("input[name='ref']").prop("disabled", false);
            }
        }
        reader.readAsText(filePath, "utf-8");
    };
    var getVerificationKey = function () {
        if ($("#email").val() != $("#emailConfirm").val()) {
            $("#emailHint").text("Two email is not same. Please check your email.").css("color", "red").show();
            return;
        }
        $("#emailHint").text("Email is sending. Please wait...").css("color", "blue").show();
        $.post("/getKey", {"email" : $("#email").val()}, function (data) {
            let dataJson = JSON.parse(data);
            if (!dataJson["status"]) {
                $("#emailHint").text(dataJson["note"]).css("color", "red").show();
            } else {
                $("#emailHint").css("color", "blue").show().text("Verification key sends to your email.");
            }
        });
    };
    var sendEmail = function () {
        if ($("#email").val() != $("#emailConfirm").val()) {
            $("#emailHint").text("Two email is not same. Please check your email.").css("color", "red").show();
            return;
        }
        let resultInputs = $("input[name='resultType']");
        if (!$("input[name='resultType']:eq(0)").prop("checked") && !$("input[name='resultType']:eq(1)").prop("checked")) {
            $("#emailHint").text("You must specify at least one result type.").css("color", "red").show();
        }
        $("#emailHint").text("Your request is sending. Please wait...").css("color", "blue").show();
        let postData = {
            "email" : $("#email").val(),
            "key" : $("#key").val(),
            "cplx" : complexData,
            "dset" : datasetData,
            "ref" : refData,
            "iter" : $("#iterInput").val(),
            "minsize" : $("#minsizeInput").val(),
            "mafdr" : $("#mafdrInput").val(),
            "scoreset" : $("#protrecscoresetInput").val(),
            "fdrset" : $("#fdrsetInput").val(),
            "onlyProtrec" : $("input[name='resultKind']:eq(0)").prop("checked"),
            "requireRawMatrix" : $("input[name='resultType']:eq(0)").prop("checked"),
            "require01Matrix" : $("input[name='resultType']:eq(1)").prop("checked"),
            "recoveryRate" : $("input[name='ref']:eq(0)").prop("checked"),
            "scoreDistribution" : $("input[name='ref']:eq(1)").prop("checked")
        };
        $.post("/sendEmail", postData, function (data) {
            let dataJson = JSON.parse(data)
            if (!dataJson["status"]) {
                $("#emailHint").text(dataJson["note"]).css("color", "red").show();
            } else {
                $("#emailHint").text("We have receive your request. Please wait for our response.").css("color", "blue").show();
            }
        });
    };
    var sendUserMessage = function () {
        if ($("#username").val() === "") {
            $("#msgHint").val("Please enter your name").css("color", "red").show();
        } else if ($("#useremail").val() === "") {
            $("#msgHint").val("Please enter your email").css("color", "red").show();
        } else if ($("#usermessage").val() === "") {
            $("#msgHint").val("Please enter your message").css("color", "red").show();
        } else {
            $("#msgHint").val("Your inquiry is sending to us... Please wait for a moment.").css("color", "blue").show();
            let postData = {
                "name" : $("#username").val(),
                "email" : $("#useremail").val(),
                "message" : $("#usermessage").val()
            };
            $.post("/sendInquiry", postData, function (data) {
                let dataJson = JSON.parse(data);
                if (!dataJson["status"]) {
                    $("#msgHint").text(dataJson["note"]).css("color", "red").show();
                } else {
                    $("#msgHint").text("We have received your inquiry.").css("color", "blue").show();
                }
            });
        }
        return false;
    };
    $(function () {
        $("#datasetSelector").val(null);
        $("#complexSelector").val(null);
        $("#refSelector").val(null);
        $("#iterInput").val("1000");
        $("#minsizeInput").val("5");
        $("#mafdrInput").val("0.01");
        $("#protrecscoresetInput").val("0.95");
        $("#fdrsetInput").val("0.05");
        $("input[name='resultKind']:eq(0)").prop("checked", true);
        $("input[name='resultType']:eq(0)").prop("checked", true);
        $("input[name='resultType']:eq(1)").prop("checked", false);
        $("input[name='ref']").prop("checked", false).prop("disabled", true);

    });
        </script>
        <style>
    .paramText {
        width: 200px;
        display: inline-block;
    }
    .paramInput {
        width : 50px;
        display: inline-block;
    }
    .hint {
        display: none;
    }
        </style>
  <!--test insert end-->
  </head>
<body class="" gtools_scp_screen_capture_injected="true">
<header id="header" class="header bg-white">
    <div class="navbar-container">
      <a href="" title="Find out more about Centre for Biomedical Informatics | NTU Singapore" class="navbar-logo"> <img src="https://cbi-ntu.github.io/CBI-Project-Website.github.io/media/logo.png" alt="Centre for Biomedical Informatics | NTU Singapore"> </a>
        <div class="navbar-menu">
            
                <a href="https://cbi-ntu.github.io/CBI-Project-Website.github.io/about">About CBI</a> 
                 
                <a href="https://cbi-ntu.github.io/CBI-Project-Website.github.io/">Our Projects</a> 
                 
                <a href="https://cbi-ntu.github.io/CBI-Project-Website.github.io/services">Our Services</a> 
                 
                <a href="https://cbi-ntu.github.io/CBI-Project-Website.github.io/contact">Contact Us</a> 
                 
          </div>
        <div class="navbar-search" onclick="">
            <span class="icon-search"></span>
            <form id="cb-search-btn" role="search">
         <span class="search-box">
         <input type="text" class="input" id="cb-search-content" required="true" placeholder="Contact Us..." maxlength="30" autocomplete="off">
        </span>
            </form>
        </div>
        <div class="navbar-mobile-menu" onclick="">
            <span class="icon-menu cross"><span class="middle"></span></span>
            <ul>
                
                <li><a href="https://cbi-ntu.github.io/CBI-Project-Website.github.io/about">About CBI</a></li> 
                 
                <li><a href="https://cbi-ntu.github.io/CBI-Project-Website.github.io/">Our Projects</a></li> 
                 
                <li><a href="https://cbi-ntu.github.io/CBI-Project-Website.github.io/services">Our Services</a></li> 
                 
                <li><a href="https://cbi-ntu.github.io/CBI-Project-Website.github.io/contact">Contact Us</a></li> 
                 
            </ul>
        </div>
    </div>
</header>
<div class="post-header-thumb bg-white}">
	<div class="post-header-thumb-op"></div>
	<div class="post-header-thumb">
		<div class="post-header-thumb-container">
    <h1 class="post-title" itemprop="name headline">
	PROTREC: a probability-based approach for recovering missing proteins based on biological networks</h1>
		 <div class="post-data">  
	 <time datetime="2022-01-01 00:00:00" itemprop="datePosted">Date Posted: 2022-01-01 00:00:00</time>
     <a href="https://cbi-ntu.github.io/CBI-Project-Website.github.io/tags#Projects" title="Find out more about Projects" data-hover="Category: Projects">Category: Projects</a>
    <a href="#read"> Reads: comments</a> 
			</div>
            <div class="post-tags">
                
      
      
       
          <a href="https://cbi-ntu.github.io/CBI-Project-Website.github.io/tags#Projects" title="Find out more aboutProjects" data-hover="Projects">
          Projects <span>(1)</span>
          
</a>
       
          <a href="https://cbi-ntu.github.io/CBI-Project-Website.github.io/tags#Bioinformatics" title="Find out more aboutBioinformatics" data-hover="Bioinformatics">
          Bioinformatics <span>(1)</span>
          
</a>
       
          <a href="https://cbi-ntu.github.io/CBI-Project-Website.github.io/tags#Protein Complexes" title="Find out more aboutProtein Complexes" data-hover="Protein Complexes">
          Protein Complexes <span>(1)</span>
          
</a>
      
      
      
     
	</div>
</div>
</div></div>
<article class="main-content post-page" itemscope itemtype="http://schema.org/Article">
	<div class="post-header">
		<h1 class="post-title" itemprop="name headline">
			PROTREC: a probability-based approach for recovering missing proteins based on biological networks</h1>
		<div class="post-data">
				<time datetime="2022-01-01 00:00:00" itemprop="datePosted">2022-01-01 00:00:00</time>
		</div>
	</div>
	<div id="post-content" class="post-content" itemprop="articleBody">
         <p class="post-tags">
    
      
      
       
          <a href="https://cbi-ntu.github.io/CBI-Project-Website.github.io/tags#Projects" title="Find out more aboutProjects" data-hover="Projects">
          Projects <span>(1)</span>
          </a>
       
          <a href="https://cbi-ntu.github.io/CBI-Project-Website.github.io/tags#Bioinformatics" title="Find out more aboutBioinformatics" data-hover="Bioinformatics">
          Bioinformatics <span>(1)</span>
          </a>
       
          <a href="https://cbi-ntu.github.io/CBI-Project-Website.github.io/tags#Protein Complexes" title="Find out more aboutProtein Complexes" data-hover="Protein Complexes">
          Protein Complexes <span>(1)</span>
          </a>
      
      
      
     
</p>
<h1 id="abstract">Abstract</h1>
<h3 id="a-novel-network-based-approach-for-predicting-missing-proteins-mps-is-proposed-here-this-approach-protrec-short-for-protein-recovery-dominates-existing-network-based-methods---such-as-functional-class-scoring-fcs-hypergeometric-enrichment-he-and-gene-set-enrichment-analysis-gsea---across-a-variety-of-proteomics-datasets-derived-from-different-proteomics-data-acquisition-paradigms-higher-protrec-scores-are-much-more-closely-correlated-with-higher-recovery-rates-of-mps-across-sample-replicates-the-protrec-score-unlike-methods-reporting-p-values-can-be-directly-interpreted-as-the-probability-that-an-unreported-protein-in-a-proteomic-screen-is-actually-present-in-the-sample-being-screened-significance-mass-spectrometry-ms-has-developed-rapidly-in-recent-years-however-an-obvious-proportion-of-proteins-is-still-undetected-leading-to-missing-protein-problems-a-few-existing-protein-recovery-methods-are-based-on-biological-networks-but-the-performance-is-not-satisfactory-we-propose-a-new-protein-recovery-method-protrec-a-bayesian-inspired-approach-based-on-biological-networks-which-shows-exceptional-performance-across-multiple-validation-strategies-it-does-not-rely-on-peptide-information-so-it-avoids-the-ambiguity-issue-that-most-protein-assembly-methods-face">A novel network-based approach for predicting missing proteins (MPs) is proposed here. This approach, PROTREC (short for PROtein RECovery), dominates existing network-based methods - such as Functional Class Scoring (FCS), Hypergeometric Enrichment (HE), and Gene Set Enrichment Analysis (GSEA) - across a variety of proteomics datasets derived from different proteomics data acquisition paradigms: Higher PROTREC scores are much more closely correlated with higher recovery rates of MPs across sample replicates. The PROTREC score, unlike methods reporting p-values, can be directly interpreted as the probability that an unreported protein in a proteomic screen is actually present in the sample being screened. SIGNIFICANCE: Mass spectrometry (MS) has developed rapidly in recent years; however, an obvious proportion of proteins is still undetected, leading to missing protein problems. A few existing protein recovery methods are based on biological networks, but the performance is not satisfactory. We propose a new protein recovery method, PROTREC, a Bayesian-inspired approach based on biological networks, which shows exceptional performance across multiple validation strategies. It does not rely on peptide information, so it avoids the ambiguity issue that most protein assembly methods face.</h3>
    <!--test insert start-->
    <span id="top"></span>
    <div id="templatemo_wrapper">
	    <div id="templatmeo_header">
            <div id="site_title">
                <h1><a href="#">PROTREC</a></h1>
            </div>
            <div id="templatemo_menu">
                <ul>
                    <li><a href="#home">Home</a></li>
                    <li><a href="#aboutus">Methods</a></li>
                    <li><a href="#example">Example</a></li>
                    <li><a href="#submit">Submit your dataset</a></li>
                    <li class="last"><a href="#submit">Contact us</a></li>
                </ul>
            </div>
	    </div>

        <div id="templatemo_main">

            <div id="home" class="content_top"></div>
            <div class="content_box">
                <div class="content_title content_home"></div>
                <h2>PROTREC</h2>
                <div class="image_wrapper">
                    <img src="static/images/abstract.png" alt="Image" style="left: 180px;"/>
                </div>
                <p><b>PROTREC is an algorithm for predicting and validating missing proteins in proteomics data based on <a href="https://www.sciencedirect.com/science/article/pii/S1874391921002918">Kong W, Wong B J H, Gao H, et al. PROTREC: A probability-based approach for recovering missing proteins based on biological networks. Journal of Proteomics, 2022, 250: 104392.</a> Here, we also include other three common used network based methods (same as in the literature) for a better comparison.</b></p>

                <div class="h30"></div>

                <div class="col_w310 float_l">
                    <h3>Functions our platform provide</h3>
                    <ul>
                        <li>Probability (or p-value) score matrix</li>
                        <li>Comparison heatmap</li>
                        <li>Recovery rate</li>
                        <li>Score distribution</li>
                    </ul>
                </div>

                <div class="col_w310 float_r">
                    <h3>Methods included in this platform</h3>
                    <ul>
                        <li> PROTREC </li>
                        <li> FCS (Functional Class Scoring) </li>
                        <li> HE (Hypergeometic Enrichment) </li>
                        <li> GSEA (Gene Set Enrichment Analysis) </li>
                    </ul>
                </div>

                <div class="cleaner"></div>
                <a href="#" class="gototop">Go To Top</a>

            </div>

            <div id="aboutus" class="content_top"></div>
            <div class="content_box">
                <div class="content_title content_methods"></div>

                <h2>Methods Description</h2>
                <div class="col_w310 float_l">
                    <h3>PROTREC</h3>
                    <p> PROTREC is a novel probability-based scoring scheme that estimates the probability of a protein being present in a screen. </p>
                    <p>Specifically, to calculate PROTREC probability for a protein \(x\), we first find all the protein complexes \(z_i\) containing protein \(x\). Not all complex we should consider since some complex contains so less proteins. <b><u>The default complex size threshold is 5, you may change it by changing `Min size of protein complex' in the parameter setting under `Run your own datasets' section.</u></b></p>
                    <p>Then, we calculate the probability of a complex \(z_i\in z\) being present:\[p(z_i)=\frac{\sum\limits_{x_i\in L}(1-FDR)}{|z_i|}\]\(x_i\) denotes a protein inside the complex \(z_i\). \(L\) denotes the set of proteins reported by the proteomic screen. If \(x_i\) is reported by the proteomics screen \(L\), then its prior probability is \((1-FDR)\), where \(FDR\) is the false discovery rate of \(L\). <u><b>The default \(FDR\) is 0.01, you may change it by changing `FDR of proteomic screen' in the parameter setting under `Run your own datasets' section.</b></u></p>
                    <p>PROTREC assume protein \(x\) being present in a sample is dependent on the joint probability of it being present if its complex is formed, and the probability it is present if its constituent complex is not formed. Since there might be multiple protein complexes, PROTREC computes the probability of a protein \(x\) being present in a sample being screened using each of the complexes that the protein \(x\) is a member of and returns the maximum:\[p(x)=\max_{z_i\in z}\{p(x|z_i)p(z_i)+p(x|\overline{z_i})p(\overline{z_i})\}\]This way, we can calculate all protein's PROTREC score. We can sort the proteins by their score and predict unreported proteins above a given PROTREC score threshold as predicted missing proteins. <b><u>By default, we use 0.95 as the cutoff. You may change your own cutoff by changing `PROTREC score threshold' in the parameter setting under `Run your own datasets' section.</u></b></p>
                </div>

                <div class="col_w310 float_r">
                    <h3>FCS</h3>
                    <p>FCS tests whether a network is significantly enriched given the observed proteins. Given a set of observed proteins in a proteomics screen \(S\), and a list of component proteins \(M\) from protein complex \(C\), an observed overlap \(O\), which is expressed as: \[O=\frac{|S\cap M|}{|M|}\]To determine if the overlap \(O\) is significant, a set \(N\) of randomized complexes of size are generated using a reference pool of unique proteins drawn from the complexes \(C\). <b><u>Default we choose \(N\) as 1000, but you may change the number by changing `Number of iterations' in the parameter setting under `Run your own datasets' section.</u></b> Among the randomized complexes, a vector of null overlaps, \(N_j\) is generated. For the \(j^{th}\) randomized complex, which comprises the set of proteins \(K_j\), \(N_j\), is defined as follows:\[N_j=\frac{|S\cap K_j|}{|K_j|}\]The empirical p-value is the proportion of null overlaps in \(N_j\) greater than or equal to the observed overlap \(O\). For the \(i^{th}\) complex \(C_i\) in the complex vector, its p-value, \(pval_i\) is:\[pval_i=\frac{\sum\limits_{j=1}^{N}[N_j\ge 0]}{N}\]If the FCS p-value falls below a significance p-value threshold, then all member proteins of the complex, including the unobserved ones, are predicted as present. </p>
                </div>

                <div class="cleaner h30"></div>

                <div class="col_w310 float_l">
                    <h3>HE</h3>
                    <p>In HE, the set of observed proteins are compared against a vector of protein complexes. Given a total number of proteins \(N\), with \(M\) of these belonging to a complex and \(n\) of these proteins in the differential set, the probability \(P\) that \(b\) or more proteins from the differential set are associated by chance with the complex is given by:\[P(X\ge b)=\sum\limits_{i=b}^{min(n,M)}\frac{C_n^iC_{N-n}^{M-i}}{C_N^M}\]\(P(X\ge b)\) is the HE p-value. A complex is declared significant if the HE p-value falls below the threshold.</p>
                </div>

                <div class="col_w310 float_r">
                    <h3>GSEA</h3>
                    <p>Gene Set Enrichment Analysis (GSEA) uses a Kolmogorov-Smirnov (KS) statistic. Here, the two-sample KS test is used to evaluate if the distribution of ranks based on the t-statistic of proteins in a complex differs from that of proteins outside the complex. Denoting proteins in the complex as the set \(C\) and proteins outside the complex as the set \(C'\), the KS-statistic \(KS_{C,C'}\)is expressed as:\[KS_{C,C'}=\max\limits_x|F_{1,C}(x)-F_{2,C'}(x)|\]where \(F_{1,C}(x)\) and \(F_{2,C'}(x)\) are respectively the fraction of proteins in \(C\) and \(C'\) whose rank is higher than the rank \(x\). The null hypothesis is rejected at a significance threshold if\[KS_{C,C'}\ge c(\alpha)*\sqrt{\frac{|C|+|C'|}{|C|*|C'|}}\]Where \(c(\alpha)\) is the critical value at a given alpha level. </p>
                </div>

                <div class="cleaner"></div>

                <div>
                    <p>Notably, <b><u>For FCS, HE and GSEA, the default p-value threshold is 0.05. You may change your own cutoff by changing `p-val cutoff for FCS, HE and GSEA' in the parameter setting under `Run your own datasets' section.</u></b></p>
                </div>

                <div class="h30"></div>
                <a href="#" class="gototop">Go To Top</a>

            </div>

            <div id="example" class="content_top"></div>
            <div class="content_box">
                <div class="content_title content_example"></div>
                <h2>Example</h2>
                <div class="col_w310 float_l">
                    <h3>Example dataset: RC_N data</h3>
                    <p>One proteomics expression datasets are provided. The renal cancer dataset (RC) comprises 12 normal (RC_N) and 12 cancer (RC_C) samples. We provide RC_N sample as an example. Press the button you may check how does the <a href="/static/csv/RCN.csv" download="RCN.csv">sample</a> looks like.</p>
                </div>
                <div class="col_w310 float_r">
                    <h3>Protein complex</h3>
                    <p>Protein complex is obtained from <a href="http://mips.helmholtz-muenchen.de/genre/proj/corum/">CORUM</a>. We use CORUM complex 2018 release human dataset as our default complex database. You may check how the complex looks like by <a href="/static/csv/complex.csv" download="complex.csv">clicking here</a></p>
                </div>
                <div class="h30 cleaner"></div>
                <h3>Example results</h3>
                <ul>
                    <li><a href="/static/csv/example_protrecres.csv" download="example_protrecres.csv">PROTREC result</a></li>
                    <li><a href="/static/csv/example_fcsres.csv" download="example_fcsres.csv">FCS result</a></li>
                    <li><a href="/static/csv/example_heres.csv" download="example_heres.csv">HE result</a></li>
                    <li><a href="/static/csv/example_gseares.csv" download="example_gseares.csv">GSEA result</a></li>
                </ul>
                <div class="h30"></div>
                <a href="#" class="gototop">Go To Top</a>
            </div>

            <div id="submit" class="content_top"></div>
            <div class="content_box">
                <div class="content_title content_submit"></div>

                <h2>Submit your dataset</h2>
                <p> You can submit your own dataset and protein complex to get the protein inference result. If it is your first time using this tool, you may refer to <a href="#">Example</a> section to see how it works. If no file is submitted, the program will use default file. We encourage users to use the default protein complex.</p>

                <div class="h30"></div>

                <div class="post_box float_l">
                    <h4>Step 1. Upload your Dataset and Protein Complex</h4>
                    <p> Dataset (Default: example dataset)</p>
                    <div>
                        <input id="datasetSelector" type="file" accept=".csv"/>
                        <div class="h10"></div>
                        <button onclick="chooseFile('datasetSelector')">Upload</button>
                        <div class="h20"></div>
                    </div>
                    <p id="datasetHint" class="hint"></p>
                    <p>Protein Complex (Default: CORUM database 2018 release)</p>
                    <div>
                        <input id="complexSelector" type="file" accept=".csv"/>
                        <div class="h10"></div>
                        <button onclick="chooseFile('complexSelector')">Upload</button>
                    </div>
                    <p id="complexHint" class="hint"></p>
                </div>

                <div class="post_box float_r">
                    <h4>Step 3. Enter your parameters</h4>
                    <div>
                        <label>Number of iterations</label><br />
                        <input id="iterInput"/>
                    </div>
                    <div>
                        <label>Min size of protein complex</label><br />
                        <input id="minsizeInput"/>
                    </div>
                    <div>
                        <label>FDR of proteomic screen</label><br />
                        <input id="mafdrInput"/>
                    </div>
                    <div>
                        <label>PROTREC score threshold</label><br />
                        <input id="protrecscoresetInput"/>
                    </div>
                    <div>
                        <label>FDR threshold for FCS, HE and GSEA</label><br/>
                        <input id="fdrsetInput"/>
                    </div>
                </div>

                <div class="cleaner h10"></div>

                <div class="post_box float_l">
                    <h4>Step 2. Select the results</h4>
					<b>Select the method</b>
					<div class="h10"></div>
                    <div>
                        <input type="radio" name="resultKind" /> <span class="optionSpan">PROTREC</span>
						&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;

                        <input type="radio" name="resultKind" /> <span class="optionSpan">ALL</span>
                    </div>
                    <div class="h20"></div>
					<b>Select the results</b>
					<div class="h10"></div>
                    <div>
                        <input type="checkbox" name="resultType" /> <span class="optionSpan">Raw predicted score matrix</span>
                        <div class="h10"></div>
                        <input type="checkbox" name="resultType" /> <span class="optionSpan">0/1 predicted score matrix</span>
                    </div>
                    <div class="h10"></div>
                    <div>
                        <input type="checkbox" name="ref" /> <span class="optionSpan">Recovery rate</span>
                        <div class="h10"></div>
                        <input type="checkbox" name="ref" /> <span class="optionSpan">Score distribution</span>
                    </div>
					<div class="h20"></div>
					<b>Note: recovery rate and score distribution can only be calculated after you submit your reference list.</b><br/>
					<div class="h10"></div>
                    Upload your reference list
                    <div class="h10"></div>
                    <div>
                        <input id="refSelector" type="file" accept=".csv"/>
                        <div class="h10"></div>
                        <button onclick="chooseFile('refSelector')">Upload</button>
                        <div class="h20"></div>
                    </div>
                    <p id = "refHint" class="hint"></p>
                </div>

                <div class="post_box float_r">
                    <h4>Step 4. Enter email and key</h4>
                    <div>
                        <label>Email Address</label> <br />
                        <input id="email"/>
                    </div>
                    <div>
                        <label>Confirm Email Address</label> <br />
                        <input id="emailConfirm"/>
                    </div>

                    <div class="h10"></div>

                    <p>Click <a href="javascript:void(0)" onclick="getVerificationKey()">here</a> to get your verification key after putting in your email address.</p>
                    <div>
                        <label>Verification Key</label> <br />
                        <input id="key"/>
                    </div>

                    <p>Click <a href="javascript:void(0)" onclick="sendEmail()">here</a> to send your data to us. We will execute your data and send results back to your email.</p>
                    <p id="emailHint" class="hint"></p>
                </div>
                <div class="cleaner"></div>
                <a href="#" class="gototop">Go To Top</a>
            </div>

            <div id="contact" class="content_top"></div>
            <div class="content_box">
                <div class="content_title content_contact"></div>
                <h2>Contact Us</h2>
                <p>We are a research group comprised of biodata scientists, computational biologists and education technologists in the School of Biological Sciences and Lee Kong Chian School of Medicine, Nanyang Technological University. </p>
                <p>Our lab is focused on the development of statistical approaches for analysing and resolving platform-specific idiosyncrasies in multi-omics data; identifying and resolving confounding issues such as batch effects, technical bias and missing values in high-dimensional data; and developing robust biomarker and drug target prediction techniques using a combination of machine learning and enhanced in silico validation techniques. Our lab is also interested in Bio-education, with an emphasis on the use of new AI-based technologies, text-mining and high-impact pedagogical practices (experiential learning), to enhance the quality of biological and biotechnological education. <a href = "https://gohwils.github.io/biodatascience/"> Here is more information about our lab.</a></p>

                <div class="h30"></div>
                <div class="col_w310 float_l">
                    <h4>Any inquiry?</h4>
                    <div id="contact_form">
                        <form>
                            <label for="username">Name:</label>
                            <input type="text" id="username" class="input_field" />
                            <div class="h10"></div>

                            <label for="useremail">Email:</label>
                            <input type="text" id="useremail" class="input_field" />
                            <div class="h10"></div>

                            <label for="usermessage">Message:</label>
                            <textarea id="usermessage" rows="0" cols="0"></textarea>
                            <div class="h10"></div>

                            <input type="button" class="submit_btn" value="Send" onclick="sendUserMessage()"/>
                        </form>
                    </div>
                    <p id="msgHint" class="hint"></p>
                    <div class="h30"></div>
                    <div class="div1"><img src="static/images/school.png" alt="Image" /></div>
                    <div class="h20"></div>
                </div>

                <div class="col_w310 float_r">
                    <h4>Our group</h4>
                    <div class="image_wrapper"><img src="static/images/lab.jpg" alt="Image" /></div>
                    <div class="h20"></div>
                    <h5>Lead PI</h5>
                    <h6> Dr. Wilson Goh</h5>
                    Assistant Professor of Biomedical Informatics, Lee Kong Chian School of Medicine <br />
                    Director, PhD Programmes, Lee Kong Chian School of Medicine <br />
                    Director, Biomedical Data Science Graduate Programme <br />
                    Co-Director, Centre for Biomedical Informatics <br />
                    Head, Good Research Practice Office <br />
                    Group Leader, Bio-Data Science and Education Laboratory <br />
                </div>
                <div class="cleaner"></div>
                <a href="#" class="gototop">Go To Top</a>
            </div>
        </div>

        <div id="templatemo_footer">
            Copyright © 2048 <a href="#">Your Company Name</a> | Template from <a href="http://lidaojia.tmall.com" target="_parent" title="网站模板">网站模板</a>
        </div>
    </div>
    <!--test insert end-->
		<!--<p class="post-info">
			本文由 <a href="/">liberxue</a> 创作，采用 <a href="https://creativecommons.org/licenses/by/4.0/" target="_blank" rel="external nofollow">知识共享署名4.0</a> 国际许可协议进行许可<br>本站文章除注明转载/出处外，均为本站原创或翻译，转载前请务必署名<br>最后编辑时间为:2022-01-01 00:00:00</p>-->
	</div>
</article>

<!-- <div class="related-post-lists">
	<div class="post-lists">
		<div class="post-lists-body">
					<div class="post-list-item">
				<div class="post-list-item-container">
					<div class="item-label">
						<div class="item-title"><a href="induction.html">归纳的随想</a></div>
						<div class="item-meta clearfix">
														<div class="item-meta-ico bg-ico-image" style="background: url('../usr/themes/pinghsu/images/bg-ico.png') no-repeat;background-size: 40px auto;"></div>
	                        							<div class="item-meta-cat"><a href="../category/thoughts/index.html">Thoughts</a></div>
						</div>
					</div>
				</div>
			</div>
				</div>
	</div>
</div> -->
<footer class="footer bg-white">
	<div class="footer-social">
		<div class="footer-container clearfix">
			<div class="social-list">
    <!--<a class="social github" target="blank" href="https://github.com/" title="Find out more about " data-hover="GitHub">GitHub</a>
    <a class="social twitter" target="blank" href="http://twitter.com/" title="Find out more about " data-hover="Twitter">Twitter</a>
    <a class="social rss" target="blank" href="/feed.xml"title="Find out more about " data-hover="RSS">RSS</a>-->
	</div>
		</div>
	</div>
	<div class="footer-meta">
        <div class="footer-container">
            <div class="meta-item meta-copyright">
                <div class="meta-copyright-info">
                    <a href="/" class="info-logo">
                         <img src="https://cbi-ntu.github.io/CBI-Project-Website.github.io/media/logo.png"   title="Find out more about LiberXue_blog" data-hover="LiberXue_blog" alt="LiberXue_blog" /> </a>
                    <div class="info-text">
                        <!--<p>Theme is <a href="http://www.liberxue.com/2017/08/03/Jekyllthemes.html" title="Find out more about Jekyll liberxue主题"  data-hover="Jekyll liberxue"target="_blank">Jekyll liberxue</a> by <a href="http://www.liberxue.com/about" target="_blank">liberxue</a></p>-->
                        <p>Powered by <a href="https://cbi-ntu.github.io/CBI-Project-Website.github.io/tags/#Jekyll" title="Jekyll" data-hover="Jekyll" target="_blank" rel="nofollow">Jekyll</a></p>
                        <p>&copy; 2017 <a href="/feed.xml"  title="CBI blog RSS" data-hover="CBI blog RSS">CBI blog RSS</a></p>
                    </div>
                </div>
            </div>
            <div class="meta-item meta-posts">
                <h3 class="meta-title">Pinned articles</h3>
     <!--置顶-->
    
      
    
      
    
      
    
        <!--置顶-->
            </div>
            <div class="meta-item meta-comments">
                <h3 class="meta-title">Latest project</h3>
          
      
 <li><a href="https://cbi-ntu.github.io/CBI-Project-Website.github.io/2022/01/01/bookindex.html" title="Find out more about PROTREC: a probability-based approach for recovering missing proteins based on biological networks" data-hover="PROTREC: a probability-based approach for recovering missing proteins based on biological networks">PROTREC: a probability-based approach for recovering missing proteins based on biological networks</a></li>
      
            
      
            
      
    
</div>
		</div>
    </div>
</footer>
<script src="//cdn.bootcss.com/jquery/2.2.2/jquery.min.js"></script>
<script src="//cdn.bootcss.com/bootstrap-3-typeahead/4.0.2/bootstrap3-typeahead.js"></script>
<script src="/search/js/cb-search.js"></script>
<!--<script>
$(function () {
    setTimeout(function () {
      $('code').removeClass('hljs');
      $('code').removeClass('language');
    }, 90);
});
(function(){
    var bp = document.createElement('script');
    var curProtocol = window.location.protocol.split(':')[0];
    if (curProtocol === 'https') {
        bp.src = 'https://zz.bdstatic.com/linksubmit/push.js';        
    }
    else {
        bp.src = 'http://push.zhanzhang.baidu.com/push.js';
    }
    var s = document.getElementsByTagName("script")[0];
    s.parentNode.insertBefore(bp, s);
})();
</script>-->

<!--liberxueconnnet-menu-->
<div id="directory-content" class="directory-content">
    <div id="directory"></div>
</div>
<!--liberxueconnnet-menu-->
<script>
var postDirectoryBuild = function() {
    var postChildren = function children(childNodes, reg) {
        var result = [],
            isReg = typeof reg === 'object',
            isStr = typeof reg === 'string',
            node, i, len;
        for (i = 0, len = childNodes.length; i < len; i++) {
            node = childNodes[i];
            if ((node.nodeType === 1 || node.nodeType === 9) &&
                (!reg ||
                isReg && reg.test(node.tagName.toLowerCase()) ||
                isStr && node.tagName.toLowerCase() === reg)) {
                result.push(node);
            }
        }
        return result;
    },
    createPostDirectory = function(article, directory, isDirNum) {
        var contentArr = [],
            titleId = [],
            levelArr, root, level,
            currentList, list, li, link, i, len;
        levelArr = (function(article, contentArr, titleId) {
            var titleElem = postChildren(article.childNodes, /^h\d$/),
                levelArr = [],
                lastNum = 1,
                lastRevNum = 1,
                count = 0,
                guid = 1,
                id = 'directory' + (Math.random() + '').replace(/\D/, ''),
                lastRevNum, num, elem;
            while (titleElem.length) {
                elem = titleElem.shift();
                contentArr.push(elem.innerHTML);
                num = +elem.tagName.match(/\d/)[0];
                if (num > lastNum) {
                    levelArr.push(1);
                    lastRevNum += 1;
                } else if (num === lastRevNum ||
                    num > lastRevNum && num <= lastNum) {
                    levelArr.push(0);
                    lastRevNum = lastRevNum;
                } else if (num < lastRevNum) {
                    levelArr.push(num - lastRevNum);
                    lastRevNum = num;
                }
                count += levelArr[levelArr.length - 1];
                lastNum = num;
                elem.id = elem.id || (id + guid++);
                titleId.push(elem.id);
            }
            if (count !== 0 && levelArr[0] === 1) levelArr[0] = 0;

            return levelArr;
        })(article, contentArr, titleId);
        currentList = root = document.createElement('ul');
        dirNum = [0];
        for (i = 0, len = levelArr.length; i < len; i++) {
            level = levelArr[i];
            if (level === 1) {
                list = document.createElement('ul');
                if (!currentList.lastElementChild) {
                    currentList.appendChild(document.createElement('li'));
                }
                currentList.lastElementChild.appendChild(list);
                currentList = list;
                dirNum.push(0);
            } else if (level < 0) {
                level *= 2;
                while (level++) {
                    if (level % 2) dirNum.pop();
                    currentList = currentList.parentNode;
                }
            }
            dirNum[dirNum.length - 1]++;
            li = document.createElement('li');
            link = document.createElement('a');
            link.href = '#' + titleId[i];
            link.title = 'Find out more about' + titleId[i];
            link.title = 'Find out more about' + titleId[i];
            link.innerHTML = !isDirNum ? contentArr[i] :
                dirNum.join('.') + ' ' + contentArr[i] ;
            li.appendChild(link);
            currentList.appendChild(li);
        }
        directory.appendChild(root);
    };
    createPostDirectory(document.getElementById('post-content'),document.getElementById('directory'), true);
};
postDirectoryBuild();
</script>
  <script src="//cdn.bootcss.com/highlight.js/9.12.0/highlight.min.js"></script>
<script >lang=hljs.initHighlightingOnLoad();</script>   
 </body>
</html>